File: src/main/java/com/projetTransversalIsi/Niveau/application/use_cases/CreateNiveauUC.java
package com.projetTransversalIsi.Niveau.application.use_cases;

import com.projetTransversalIsi.Niveau.application.dto.CreateNiveauRequestDTO;
import com.projetTransversalIsi.Niveau.domain.Niveau;

public interface CreateNiveauUC {
    Niveau execute(CreateNiveauRequestDTO request);
}

File: src/main/resources/static/js/NiveauJs/application/CreateNiveauUC.js
export class CreateNiveauUC {
    constructor(api) {
        this.api = api;
    }

    async execute(request) {
        if (!request.filiereId) {
            throw new Error('L\'ID de la filière est requis');
        }
        if (!request.ordre) {
            throw new Error('L\'ordre du niveau est requis');
        }
        return await this.api.create(request);
    }
}
